src/decoder/wav.rs

```diff
@@ -65,36 +76,70 @@ impl<R> Iterator for SamplesIterator<R>
 where
     R: Read + Seek,
 {
-    type Item = i16;
+    type Item = DecoderSample;
 
     #[inline]
-    fn next(&mut self) -> Option<i16> {
+    fn next(&mut self) -> Option<Self::Item> {
+        self.samples_read += 1;
         let spec = self.reader.spec();
-        match (spec.sample_format, spec.bits_per_sample) {
-            (SampleFormat::Float, 32) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                f32_to_i16(value.unwrap_or(0.0))
-            }),
-            (SampleFormat::Int, 8) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                i8_to_i16(value.unwrap_or(0))
-            }),
-            (SampleFormat::Int, 16) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                value.unwrap_or(0)
-            }),
-            (SampleFormat::Int, 24) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                i24_to_i16(value.unwrap_or(0))
-            }),
-            (SampleFormat::Int, 32) => self.reader.samples().next().map(|value| {
-                self.samples_read += 1;
-                i32_to_i16(value.unwrap_or(0))
-            }),
-            (sample_format, bits_per_sample) => {
-                panic!("Unimplemented wav spec: {sample_format:?}, {bits_per_sample}")
-            }
-        }
+        let next_sample: Option<Self::Item> =
+            match (spec.sample_format, spec.bits_per_sample as u32) {
+                (SampleFormat::Float, bits) => {
+                    if bits == 32 {
+                        let next_f32: Option<Result<f32, _>> = self.reader.samples().next();
+                        next_f32.and_then(|value| {
+                            let value = value.ok();
+                            #[cfg(feature = "integer-decoder")] // perf
+                            let value = value.map(|value| value.to_sample());
+                            value
+                        })
+                    } else {
+                        // > 32 bits we cannot handle, so we'll just return equilibrium
```

Comment:
do we want to log or print here? Returning equilibrium is a silent error and will be very hard to debug. I would be okay with panicking too.